frame/vamm/src/lib.rs

```diff
@@ -737,14 +748,21 @@ pub mod pallet {
 			// Vamm must exist.
 			let mut vamm_state = Self::get_vamm_state(&vamm_id)?;
 
-			// Delegate update twap to internal functions.
+			// Other sanity checks must pass.
+			Self::update_twap_sanity_check(&vamm_state, base_twap, quote_twap, &None)?;
+
+			// Handle optional values.
 			let (base_twap, quote_twap) = match (base_twap, quote_twap) {
-				(Some(base_twap), Some(quote_twap)) =>
-					Self::do_update_twap(vamm_id, &mut vamm_state, base_twap, quote_twap, &None)?,
-				_ => Self::update_vamm_twap(vamm_id, &mut vamm_state, &None)?,
+				(Some(base_twap), Some(quote_twap)) => (base_twap, quote_twap),
+				(Some(base_twap), None) => (base_twap, Self::reciprocal_twap(&base_twap)?),
+				(None, Some(quote_twap)) => (Self::reciprocal_twap(&quote_twap)?, quote_twap),
+				(None, None) => Self::compute_new_twaps(&vamm_state, &None)?,
```

Comment:
Due to the complexity of input handling and given the fact that `FixedPointNumber` already has a convenient method to compute `reciprocal`s, I would consider restricting the function signature to take in only the `base_twap`.
This would simplify the input parsing and wouldn't burden the caller too much, since it may use `.reciprocal()` if it has the quote TWAP in hand before passing it into this function.

Reply:
With this, you could also have one less error checking, since `TwapsMustBeReciprocals` wouldn't be needed with this change

Reply:
Oh, yeah changing the function to take only `base_twap` really helps and simplifies a **lot**!
Thanks for this suggestion :)

I'm more prone to keep `TwapsMustBeReciprocals` though, because in the function `do_update_twap` we take both `base_twap` and `quote_twap`. There I think it's still important to have sanity checks before updating the runtime storage, as we don't want to modify the runtime with incorrect values by any means. (it would be just a matter of more caution actually, since these errors should not occur at all due to the way we're using the functions in the pallet.)